src/polygon.c

```diff
@@ -850,6 +852,62 @@ spherepoly_in(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(poly);
 }
 
+Datum
+spherepoly_deg(PG_FUNCTION_ARGS)
+{
+	int num_elements, i, point_index, np;
+	ArrayType *float_vector;
+	float8 *array_data;
+	SPOLY *spoly;
+	SPoint *points;
+
+	float_vector = PG_GETARG_ARRAYTYPE_P(0);
+	num_elements = ArrayGetNItems(
+		ARR_NDIM(float_vector), ARR_DIMS(float_vector)
+	);
+	np = num_elements / 2;
+
+	if (num_elements <= 0)
+	{
+		elog(ERROR, "spherepoly_deg: you need pass at least 6 floats");
+		PG_RETURN_NULL();
+	}
+
+	if (!(num_elements >= 6) || !(num_elements % 2 == 0))
```

Comment:
I propose to simplify the logical expression: !(num_elements >= 6)     ->   num_elements < 6